### PR TITLE
WIP: Test webhook button

### PR DESCRIPTION
**Pull request by wilhelmklopp**
## What does this PR do?
This adds a "Test Webhook" button that our developers can use to send a test webhook. This is especially useful on weekends where developers wouldn't be seeing many "example" webhooks come through that they can build their API around.

You can see what it looks like by creating a webhook, and putting a remix of this project as the webhook URL: https://glitch.com/edit/#!/remix/uclapi-webhooks-testing

Tests to come

## ✅ Pull Request checklist

- [X] Is this code complete?
- [ ] Are tests done/modified?

## 🚨 Is this a breaking change for API clients?
No

## :squirrel: Deploy notes
Nothing special

Depends on #187

### CODE REVIEW

**Comment by ChristopherHammond13** [comment]
Build finished. 
Jenkins statics successfully built.

**Comment by ChristopherHammond13** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://jenkins.uclapi.com/job/UCLAPI/123/
Jenkins statics successfully built.

File: backend/uclapi/dashboard/tests.py [hunk 1]
```diff
@@ -324,6 +324,21 @@ def test_edit_webhook_POST_user_owns_app_changing_url_verification_ok(
...
+    #     request = self.factory.post(
+    #         '/',
+    #         {
+    #             'app_id': self.app1.id, 'siteid': 2, 'roomid': 2,
+    #             'contact': 2, 'url': "http://new"
+    #         }
+    #     )
+    #     request.session = {'user_id': self.user1.id}
+    #
+    #     response = test_webhook(request)
```
Comment (ChristopherHammond13) [inline backend/uclapi/dashboard/tests.py]: Any reason for these commented patches? Should we keep that code or remove it?

File: backend/uclapi/dashboard/tests.py [hunk 2]
```diff
@@ -442,3 +457,6 @@ def test_refresh_verification_secret_POST_success(
         self.assertEqual(response.status_code, 200)
         self.assertTrue(content["ok"])
         self.assertTrue("new_secret" in content.keys())
+
+
+class RefreshVerifcationSecretViewTests(TestCase):
```
[the remark this answers is not in the document]
Comment (ChristopherHammond13) [reply backend/uclapi/dashboard/tests.py]: This test either needs finishing or removing as the Python code will not parse or run.

File: backend/uclapi/dashboard/views.py [hunk 3]
```diff
@@ -147,19 +147,43 @@ def dashboard(request):
...
+    # except KeyError:
+    #     logged_in = False
+    #
+    # temp_token = TemporaryToken.objects.create()
+    # return render(request, 'getStarted.html', {
+    #     'initial_data': {
+    #         'temp_token': temp_token.api_token,
+    #         'logged_in': str(logged_in)
+    #     }
+    # })
```
Comment (ChristopherHammond13) [inline backend/uclapi/dashboard/views.py]: I'm not quite sure what's going on here

File: backend/uclapi/dashboard/views.py [hunk 4]
```diff
@@ -147,19 +147,43 @@ def dashboard(request):
...
+        "state": "best-state",
+        "scopes": s.scope_dict(2),
+        "user": {
+            "full_name": "Wilhelm Klopp",
+            "cn": "zceiwhs",
+            "email": "will@ucl.ac.uk",
+            "department": "School of Management",
+            "upi": "WHSKL66"
+        },
+        "signed_data": "signed803458903459803045"
```
Comment (ChristopherHammond13) [inline backend/uclapi/dashboard/views.py]: Not sure how the data signing works. Is this value static for that page data or will it need to be generated at run time?

File: frontend/src/sass/pages/dashboard.scss [hunk 5]
```diff
@@ -28,6 +28,10 @@ body {
   background-color: rgba(234, 237, 241, 1);
 }
 
+input {
+  color: #000;
```
Comment (ChristopherHammond13) [inline frontend/src/sass/pages/dashboard.scss]: I'm fine with this as long as there are no knock-on effects of making this change global to the whole dashboard.

File: backend/uclapi/dashboard/views.py (hunk 4 above)
Comment (anirudhpillai) [reply backend/uclapi/dashboard/views.py]: page_data doesn't have logged_in as a key, value, which is used by the nav bar on the get started page to check whether to show a 'log in' button or not.

**Comment by ChristopherHammond13** [comment]
I'm assuming that this code is stale now?

**Comment by ChristopherHammond13** [comment]
I'm closing this because we'll launch a test Webhook button when we launch a new Dashboard at some point in the coming future.

**Pull request closed without merging**